apps/admin-portal/src/api/application.ts

```diff
@@ -211,3 +211,144 @@ export const getApplicationList = (limit: number, offset: number): Promise<any>
+                return Promise.reject(new Error("Inbound protocol not configured"));
+            } else if (response.status !== 200) {
+                return Promise.reject(new Error("Failed get OIDC meta data from: "));
+            }
+            return Promise.resolve(response.data as OIDCMetadataInterface);
+        }).catch((error) => {
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Gets the application's OIDC data.
+ *
+ * @param id Application ID
+ */
+export const getOIDCData = (id: string): Promise<any> => {
+    const requestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: ServiceResourcesEndpoint.applications + "/" + id + "/inbound-protocols/oidc"
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Failed retrieve OIDC data from: "));
+            }
+            return Promise.resolve(response.data as OIDCdataInterface);
+        }).catch((error) => {
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Updates the OIDC configuration.
+ *
+ * @param id Application ID
+ * @param OIDC OIDC configuration data.
+ */
+export const updateOIDCData = (id: string, OIDC: object): Promise<any> => {
+    const requestConfig = {
+        data: OIDC,
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.PUT,
+        url: ServiceResourcesEndpoint.applications + "/" + id + "/inbound-protocols/oidc"
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Failed update inbound configuration"));
+            }
+            return Promise.resolve(response);
+        }).catch((error) => {
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Updates the application configuration.
+ *
+ * @param id Application ID
+ * @param advancedConfigs Application's advanced configurations.
+ */
+export const updateAdvanceConfigurations = (id: string, advancedConfigs: object): Promise<any> => {
+    const requestConfig = {
+        data: advancedConfigs,
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": GlobalConfig.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.PATCH,
+        url: ServiceResourcesEndpoint.applications + "/" + id
+    };
+
+    return httpClient.request(requestConfig)
+        .then((response) => {
+            if (response.status !== 200) {
```

Comment:
Remove the if condition. Please check in the other places as well.

Reply:
Same as mentioned above